httpie/uploads.py

```diff
@@ -64,13 +67,53 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+def is_stdin(file: IO) -> bool:
+    try:
+        file_no = file.fileno()
+    except Exception:
+        return False
+    else:
+        return file_no == sys.stdin.fileno()
+
+
+READ_THRESHOLD = float(os.getenv("HTTPIE_STDIN_READ_THRESHOLD", 10.0))
+
+
+def trigger_stdin_warning_thread(env: Environment, file: IO) -> None:
+    # Windows unfortunately does not support select() operation
+    # on regular files, like stdin in our use case.
+    # https://docs.python.org/3/library/select.html#select.select
+    if os.name == 'nt':
+        return None
+
+    import select
+    import threading
+
+    def worker():
+        can_read, _, _ = select.select([file], [], [], READ_THRESHOLD)
+        if not can_read:
+            env.stderr.write(
+                f"> no stdin data read in {READ_THRESHOLD}s "
+                f"(perhaps you want to --ignore-stdin)\n"
+            )
+            env.stderr.write("> https://httpie.io/docs/cli/best-practices\n")
```

Comment:
We should probably use the existing `env.log_error(msg, level='warning')`
https://github.com/httpie/httpie/blob/245cede2c274bef7b172fe65814bfabc6f87b0b4/httpie/context.py#L137-L139

Reply:
And the link could be prefixed with `See: `

Reply:
@jakubroztocil any idea why `log_error` chooses to write to the original stderr instead of the regular one (which becomes devnull on `-qq`). If I switch, then it prints the warnings even with `-qq`.

Reply:
I vaguely remember there being one, likely related to  `HTTPieArgumentParser._setup_standard_streams()`:

https://github.com/httpie/httpie/blob/17ed3bb8c519fc6e0cc62fa3bb0fc10929de2595/httpie/cli/argparser.py#L197-L235

Take a look. Otherwise, it would be the natural approach 👍🏻

Reply:
Seems like some tests (on the download logic) was depending on this behaviour. Currently leaving as is and printing to `stderr` directly for now (to support `-qq`).